### PR TITLE
Save Deposit Cache to Disk

### PR DESCRIPTION
**Pull request by nisdas**
- [x] Define our deposit containers as a proto object.
- [x] Save all deposit containers to disk, so that on startup we fill our caches 
with those containers.

### CODE REVIEW

File: beacon-chain/cache/depositcache/deposits_cache.go [hunk 1]
```diff
@@ -8,6 +8,10 @@ import (
 	"sort"
 	"sync"
 
+	"github.com/prysmaticlabs/prysm/shared/bytesutil"
+
+	dbpb "github.com/prysmaticlabs/prysm/proto/beacon/db"
+
```
Comment (rauljordan) [inline beacon-chain/cache/depositcache/deposits_cache.go]: Spacings

File: beacon-chain/cache/depositcache/deposits_cache.go [hunk 2]
```diff
@@ -78,11 +73,33 @@ func (dc *DepositCache) InsertDeposit(ctx context.Context, d *ethpb.Deposit, blo
...
+	sort.SliceStable(ctrs, func(i int, j int) bool { return ctrs[i].Index < ctrs[j].Index })
+	dc.deposits = ctrs
+	historicalDepositsCount.Add(float64(len(ctrs)))
+}
+
+// AllDepositContainers returns a list of deposits all historical deposit containers until the given block number.
+func (dc *DepositCache) AllDepositContainers(ctx context.Context) []*dbpb.DepositContainer {
+	ctx, span := trace.StartSpan(ctx, "BeaconDB.AllDepositContainers")
+	defer span.End()
+	dc.depositsLock.RLock()
```
Comment (rauljordan) [inline beacon-chain/cache/depositcache/deposits_cache.go]: hmm do we need to do this...? wont the defer happen right before the return?

File: beacon-chain/cache/depositcache/deposits_test.go [hunk 3]
```diff
@@ -6,6 +6,8 @@ import (
 	"math/big"
 	"testing"
 
+	dbpb "github.com/prysmaticlabs/prysm/proto/beacon/db"
```
Comment (rauljordan) [inline beacon-chain/cache/depositcache/deposits_test.go]: spacing

File: beacon-chain/cache/depositcache/deposits_cache.go (hunk 2 above)
Comment (nisdas) [reply beacon-chain/cache/depositcache/deposits_cache.go]: defer happens after the value is returned, so we would still need this

**Comment by prestonvanloon** [comment]
Please fix bazel imports for docker images. Sorry, gazelle doesn't do it for you :(

**Pull request merged**